docs/src/explanations/connectable.md

```diff
@@ -0,0 +1,444 @@
+
+## Waived Data
+
+It is not uncommon for a user to want to connect Chisel components which are not type equivalent. For example, a user may want to cook up the `ready`/`valid` members of a `ReadyValidIO` to a `DecoupledIO`, but
+because the `bits` member is not present in both, our operators would reject a connection.
+
+`WaivedData` is the mechanism to specialize connection operator behavior in these scenarios. For any addition member which is not present in the other component being connected to, they can be explicitly waived from the operator to be ignored, rather than trigger an error.
+
+### Connecting sub-types to super-types by waiving extra members
+
+In the following example, we can use `:<>=` to connect a `MyReadyValid` to a `MyDecoupled` by waiving the `bits` member.
+
+```scala mdoc:silent
+class MyReadyValid extends Bundle {
+  val valid = Bool()
+  val ready = Flipped(Bool())
+}
+class MyDecoupled extends MyReadyValid {
+  val bits = UInt(32.W)
+}
+class Example5 extends RawModule {
+  val in  = IO(Flipped(new MyDecoupled()))
+  val out = IO(new MyReadyValid())
+  out :<>= in.waive(_.bits)
+}
+```
+
+This generates the following Verilog, where `ready` and `valid` are connected, and `bits` is ignored:
+
+```scala mdoc:verilog
+import chisel3.stage.ChiselStage
+
+ChiselStage.emitVerilog(new Example5())
+```
+
+### Connecting types with optional members
+
+In the following example, we can use `:<>=` and `waive` to connect two `MyDecoupledOpts`'s, where only one has a `bits` member.
+
+```scala mdoc:silent
+class MyDecoupledOpt(hasBits: Boolean) extends Bundle {
+  val valid = Bool()
+  val ready = Flipped(Bool())
+  val bits = if (hasBits) Some(UInt(32.W)) else None
+}
+class Example6 extends RawModule {
+  val in  = IO(Flipped(new MyDecoupledOpt(true)))
+  val out = IO(new MyDecoupledOpt(false))
+  out :<>= in.waive(_.bits.get) // We can know to call .get because we can inspect in.bits.isEmpty
+}
+```
+
+This generates the following Verilog, where `ready` and `valid` are connected, and `bits` is ignored:
+
+```scala mdoc:verilog
+import chisel3.stage.ChiselStage
+
+ChiselStage.emitVerilog(new Example6())
+```
+
+### Connecting different sub-types
+
+Note that the connection operator requires the `consumer` and `producer` to be the same Scala type to encourage capturing more information statically, but they can always be cast to `Data` or another common supertype prior to connecting.
+
+In the following example, we can use `:<>=` and `waiveAs` to connect two different sub-types of `MyReadyValid`.
+
+```scala mdoc:silent
+class HasBits extends MyReadyValid {
+  val bits = UInt(32.W)
+}
+class HasEcho extends MyReadyValid {
+  val echo = Flipped(UInt(32.W))
+}
+class Example7 extends RawModule {
+  val in  = IO(Flipped(new HasBits()))
+  val out = IO(new HasEcho())
+  out.waiveAs[MyReadyValid](_.echo) :<>= in.waiveAs[MyReadyValid](_.bits)
```

Comment:
This example is really confusing to me. Is it ignoring `echo` because you told it to waive it or because `MyReadyValid` doesn't even have an `echo` field? Would this be equivalent, if not what is the point of the `MyReadyValid` in the brackets, you didn't have it in your first example.

`out.waiveAs[MyReadyValid]() :<>= in.waiveAs[MyReadyValid](_.bits)`